code/Player/Grub/Grub.Death.cs

```diff
@@ -147,6 +149,32 @@ private void FinishDie()
 		}
 	}
 
+	private void IncrementKilledStats()
+	{
+		var attacker = GamemodeSystem.Instance.ActivePlayer;
+		if ( attacker.Client.IsBot ) return;
+
+		string statIdent = "grubs-killed";
+		if ( attacker == Player ) // Killed self.
+			statIdent = "own-grubs-killed";
+		else if ( Player.Client.IsBot ) // Killed a bot.
+			statIdent = "bot-grubs-killed";
+
+		Sandbox.Services.Stats.Increment( attacker.Client, statIdent, 1 );
+
+		// Get weapon used to kill this grub.
+		Weapon weapon = null;
+		if ( DeathReason.FirstInfo.HasValue && DeathReason.FirstInfo.Value.Weapon is Weapon )
+			weapon = DeathReason.FirstInfo.Value.Weapon as Weapon;
+		if ( weapon == null && DeathReason.SecondInfo.HasValue && DeathReason.SecondInfo.Value.Weapon is Weapon )
+			weapon = DeathReason.SecondInfo.Value.Weapon as Weapon;
+
+		if ( !weapon.IsValid() ) return;
+
+		if ( weapon.Name == "Bitch Slap" )

```

Comment:
Not a huge priority, but I would like to get one stat for every weapon.

We can slug the name of the weapon to lower-dashed-case, and then format it into the stat name `weapon-{slug}-kills`.